app/controllers/price_policies_controller.rb

```diff
@@ -6,193 +6,176 @@ class PricePoliciesController < ApplicationController
-    render 'price_policies/index'
+    render "price_policies/index"
   end
 
-  # GET /price_policies/new
+  # GET /facilities/:facility_id/{product_type}/:product_id/price_policies/new
   def new
-    # If there are active policies, start tomorrow. If none, start today
-    @start_date     = Date.today + (@product.price_policies.current.empty? ? 0 : 1)
-
-    @expire_date    = PricePolicy.generate_expire_date(@start_date)
-    @max_expire_date = @expire_date
-
-    build_price_policies
-
-    # Base the new policies off the most recent version
-    new_price_policy_list = []
-    @price_policies.each do |pp|
-      existing_pp = @product.price_policies.where(:price_group_id => pp.price_group.id).order(:expire_date).last
-      new_price_policy_list << (existing_pp ? existing_pp.dup : pp)
-    end
-    # If it's all new policies (i.e. nothing changed in the list), make the default can_purchase true
-    new_price_policy_list.each { |pp| pp.can_purchase = true } if @price_policies == new_price_policy_list
-
-    @price_policies = new_price_policy_list
-
-    render 'price_policies/new'
+    @start_date = new_start_date
+    @expire_date = set_max_expire_date
+    build_price_policies!
+    base_new_policies_on_most_recent
+    render "price_policies/new"
   end
 
-  # POST /price_policies
+  # POST /facilities/:facility_id/{product_type}/:product_id/price_policies
   def create
-    @start_date = start_date_from_params
-    @expire_date   = params[:expire_date]
-    build_price_policies
-    update_policies_from_params
-
-    respond_to do |format|
-      if ActiveRecord::Base.transaction do
-          raise ActiveRecord::Rollback unless @price_policies.all?(&:save)
-          flash[:notice] = 'Price Rules were successfully created.'
-          format.html { redirect_to facility_product_price_policies_path }
-        end
-      else
-        flash[:error] = "There was an error saving the policy"
-        format.html { render "price_policies/new" }
-      end
-    end
+    create_or_update(:new)
   end
 
-  # GET /price_policies/1/edit
-  def edit
-    @start_date = start_date_from_params
+  # PUT /facilities/:facility_id/{product_type}/:product_id/price_policies/:id
+  def update
+    create_or_update(:edit)
+  end
 
-    build_price_policies
+  # GET /facilities/:facility_id/{product_type}/:product_id/price_policies/:id/edit
+  def edit
+    @expire_date = @price_policies.map(&:expire_date).compact.first
 
-    @expire_date=@price_policies.map(&:expire_date).compact.first
-    @max_expire_date = PricePolicy.generate_expire_date(@start_date)
-    render 'price_policies/edit'
+    render "price_policies/edit"
   end
 
-  # PUT /price_policies/1
-  def update
-    @start_date = start_date_from_params
+  # DELETE /facilities/:facility_id/{product_type}/:product_id/price_policies/:id
+  def destroy
+    return remove_active_policy_warning if @start_date <= Date.today
```

Comment:
Can we call this method something like `remove_active_policy_warning_and_redirect` or something at implies that it does everything that needs to happen in a response?